### ISSUE TITLE
Notifications timeline

### ISSUE THREAD

**Issue by jdenen** [martianhi/mastodon.el#37]
Add a notifications timeline and bind it to `N`.

**Comment by alexjgriffith** [comment]
Working on the rendering of four types of notifications
1. mention
2. follow
3. favourite
4. reblog (boost)

Mention, favourite, and boost are very similar to standard toots. Follows will have a tag line (which could be customizable) and will insert the image of the new follower before the end of the notification. 
## Mention
```
$content
 | $display_name (@$user_name) Mentioned $your_display_name(@$your_user_name)
  ------------
```

## Follow
```
Congratulations, you have a new follower!
$user_image
 | $display_name (@$user_name) Following $your_display_name(@$your_user_name)
  ------------
```
## Favourite
```
$content
 | $display_name (@$user_name)  Favorited $your_display_name(@$your_user_name)
  ------------
```

## Booseted
```
$content
 | $display_name (@$user_name)  Boosted $your_display_name(@$your_user_name)
  ------------
```

The functionality will be in the `mastodon-notifications` module and will depend explicitly on `mastodon-media` and `mastodon-http`

A lot of the utility functions overlap with those in `mastdodon-tl`. For the time being I am recreating the functionality in this module. In the future we can refactor the shared functionality into a separate module.

**Comment by alexjgriffith** [comment]
Required before full implementation of streams #164

**Comment by alexjgriffith** [comment]
Closed by #178
